edb/schema/objects.py

```diff
@@ -1329,24 +1252,174 @@ def as_shell(self, schema: s_schema.Schema) -> ObjectShell:
             schemaclass=type(self),
         )
 
+    def get_ddl_identity(
+        self,
+        schema: s_schema.Schema,
+    ) -> Optional[Dict[str, str]]:
+        ddl_id_fields = [
+            fn for fn, f in type(self).get_fields().items() if f.ddl_identity
+        ]
+
+        ddl_identity: Optional[Dict[str, Any]]
+        if ddl_id_fields:
+            ddl_identity = {}
+            for fn in ddl_id_fields:
+                v = self.get_field_value(schema, fn)
+                if v is not None:
+                    ddl_identity[fn] = v
```

Comment:
Does None mean default (= insignificant for identity) for all fields?

Reply:
`None` means there's no information for this identity field at the moment.  I need to see if this can be turned into a hard error.